castle.go

```diff
@@ -9,29 +9,38 @@ import (
 	"github.com/tomasen/realip"
 )
 
-// TrackEndpoint defines the tracking URL castle.io side
-var TrackEndpoint = "https://api.castle.io/v1/track"
+// FilterEndpoint defines the filter URL castle.io side
+var FilterEndpoint = "https://api.castle.io/v1/filter"
 
-// AuthenticateEndpoint defines the adaptive authentication URL castle.io side
-var AuthenticateEndpoint = "https://api.castle.io/v1/authenticate"
+// RiskEndpoint defines the risk URL castle.io side
+var RiskEndpoint = "https://api.castle.io/v1/risk"
 
-// Event is an enum defining types of event castle tracks
-type Event string
+type Event struct {
+	EventType   EventType
+	EventStatus EventStatus
+}
+
+// EventType is an enum defining types of event castle tracks
+type EventType string
+
+// See https://docs.castle.io/docs/events
+const (
+	EventTypeLogin                EventType = "$login"
+	EventTypeRegistration         EventType = "$registration"
+	EventTypeProfileUpdate        EventType = "$profile_update"
+	EventTypeProfileReset         EventType = "$profile_reset"
+	EventTypePasswordResetRequest EventType = "$password_reset_request"
+	EventTypeChallenge            EventType = "$challenge"
+)
+
+// EventStatus is an enum defining the statuses for a given event.
+type EventStatus string
 
-// See https://castle.io/docs/events
+// See https://docs.castle.io/docs/events
```

Comment:
Not sure this doc helps with understanding the statuses

Reply:
In the events table it mentions the different statuses that are supported for events. 
e.g for $challenge:
```
A user is sent through an additional verification flow, such as Two-factor or Email-verification. This event supports the status $requested in addition to $failed and $succeeded.
```

Reply:
Yeah, I know. But it does not list what statuses are available for what types.

Reply:
Maybe a link to the reference? (https://reference.castle.io/)